Add initial sensor dictionary with CAN IDs

AdamBusch opened the pull request:
Add ability to filter out each specific sensor that is being sent over the wireless server. This will be needed when we start to actually parse BIG DATA.

AdamBusch committed:
Add initial sensor dictionaly with CAN IDs

This is a somewhat complete list of snesors on the DCAN bus that will be
sent over the wireless telemetry server. We need to add additional
sensors on the DCAN such as BMS info and finalize the IDs for each
sensor.

AdamBusch committed:
Add potential IDs and byte indicies.

First revision of CAN information for new sensors

AdamBusch committed:
Update message ID formatting

AdamBusch committed:
Remove client HTML code

Static HTML client is no longer needed for viewing data

AdamBusch committed:
Update tire temp byte length

Ture temp is being sent as a double value, needs 8 bits to represent
it.

jacklecroy committed:
Adds a function to read sensor data from dictionary

data_to_dict Takes in data and reformats it to make it more readable. As long as the SensorDict.csv stays in the same format, this probably won't need many changes in the future.

AdamBusch committed:
Add print formatting for excel/google sheets

Adds an option for a console printout of the CSV file to be pasted into
the excel sheet that keeps track of all the CAN IDs

jacklecroy committed:
Adds function to save sensor data from a list

Also updated data_to_dict to include "timestamp" in its output. This is so that these functions can be used together, both using timestamps.

jacklecroy committed:
Merge branch 'feature_add_sensor_translation' of https://github.com/PurdueElectricRacing/Wireless-Telemetry-Server into feature_add_sensor_translation

AdamBusch committed:
Output correct start/stop bits for can IDs

jacklecroy committed:
Update SensorDict.py to coding standards

Used autopep8 to update the file.

jacklecroy committed:
Updates SensorDict.py with line limit 80

Limit was not 80 when I ran autopep before.

AdamBusch committed:
Merge branch 'feature_add_sensor_translation' of https://github.com/PurdueElectricRacing/Wireless-Telemetry-Server into feature_add_sensor_translation

AdamBusch committed:
Change CAN message format for IMU

AdamBusch committed:
Fix linting issues

Travis plz

AdamBusch committed:
Add support for parsing multiple data points

Will import data from 'SensorData.csv' and parse incoming data packets
based on the information peovided in the file. This file is used to
define each sensor's address and start/end bytes used across the CAN bus
and wireless Telemetry server.

RyanSchw commented on SensorDict.py:
```diff
@@ -0,0 +1,70 @@
+        if (row["id"] != id):
+            continue
+        lsb = int(row["LSB"])
+        # Least significant byte
```
I think this can be written better, see the [pep8](https://www.python.org/dev/peps/pep-0008/#comments) guide for comments. I think comments should either be inline or above the statement. We could also rename the variables to make it more clear

AdamBusch committed:
Split sensors to front/back of car

Sensors on the same CAN ID cannot be sent from oppisite sides of the
car, need to be grouped into which MC is sending their data (front/back)

Encapsulate CSV reading into class

This will allow for us to pre-load the sensor dict to speed up parsing
individual data frames.

Re-arrange dict format

MSB should be places before LSB as this is how it is formatted in the
CAN messages.

Add sample usage code and doccumentation

Sample code can be used as a quick unit test and to ensure that the CSV
format is valid

AdamBusch committed:
Merge branch 'feature_add_sensor_translation' of https://github.com/PurdueElectricRacing/Wireless-Telemetry-Server into feature_add_sensor_translation

AdamBusch committed:
Fix linting issues

AdamBusch committed:
Fix typos mentioned in #7

bsmrdel reviewed (approved):
Looks good. We just need to make sure that people working on sensors are aware this CAN ID dictionary exists and we're all using the same IDs for the same BIG DATA

RyanSchw commented:
Can I delete the `feature_add_sensor_translation` branch now?
